WordPress/gutenberg
# Add `default-spacing-sizes` and `default-font-sizes` options for classic themes

## What?


Fixes an issue where defaults were generated in classic themes when they should not have been.

## Why?


Bug fix for https://github.com/WordPress/gutenberg/pull/58409 and https://github.com/WordPress/gutenberg/pull/61842.

## How?


Set `defaultFontSizes` and `defaultSpacingSizes` to `false` for classic themes when they have `fontSizes` or `spacingSizes` defined, respectively.

Additionally, since it was just a couple extra lines, I added the `editor-spacing-sizes`, `default-font-sizes`, and `default-spacing-sizes` theme supports for parity with other presets.

## Testing Instructions





1. In a classic theme add theme support for `editor-font-sizes` that match the default slugs. (The default slugs are: `small`, `medium`, `large`, `x-large`, and `xx-large`.)
3. See that the default font sizes are not generated, and the slugs that matched the default slugs show the theme support configuration.
4. Add theme support for `default-font-sizes` and see that the default font sizes are generated and they override the theme font sizes.

```php
function my_theme_setup() {
	// Step 1
	add_theme_support(
		'editor-font-sizes',
		array(
			array(
				'name'      => __( 'Small' ),
				'size'      => 19.5,
				'slug'      => 'small',
			),
			array(
				'name'      => __( 'Normal' ),
				'size'      => 22,
				'slug'      => 'normal',
			),
			array(
				'name'      => __( 'Large' ),
				'size'      => 36.5,
				'slug'      => 'large',
			),
			array(
				'name'      => __( 'Huge' ),
				'size'      => 49.5,
				'slug'      => 'huge',
			),
		)
	);
	// Step 2
	add_theme_support( 'default-font-sizes' );
}
add_action( 'after_setup_theme', 'my_theme_setup' );
```

```scss
:root {
	// Step 1
	--wp--preset--font-size--small: 19.5px;
	--wp--preset--font-size--medium: 20px;
	--wp--preset--font-size--large: 36.5px;
	--wp--preset--font-size--x-large: 42px;
	--wp--preset--font-size--normal: 22px;
	--wp--preset--font-size--huge: 49.5px;

	// Step 2
	--wp--preset--font-size--small: 13px;
	--wp--preset--font-size--medium: 20px;
	--wp--preset--font-size--large: 36px;
	--wp--preset--font-size--x-large: 42px;
	--wp--preset--font-size--normal: 22px;
	--wp--preset--font-size--huge: 49.5px;
}
```

The same should work for `editor-spacing-sizes` and `default-spacing-sizes`.

```php
function my_theme_setup() {
	// Step 1
	add_theme_support(
		'editor-spacing-sizes',
		array(
			array(
				'name' => __( '3X-Small' ),
				'size' => '10px',
				'slug' => '10',
			),
			array(
				'name' => __( 'Medium' ),
				'size' => '50px',
				'slug' => '50',
			),
			array(
				'name' => __( 'Large' ),
				'size' => '60px',
				'slug' => '60',
			),
			array(
				'name' => __( '3X-Large' ),
				'size' => '90px',
				'slug' => '90',
			),
		)
	);
	// Step 2
	add_theme_support( 'default-spacing-sizes' );
}
add_action( 'after_setup_theme', 'my_theme_setup' );
```

```scss
:root {
	// Step 1
	--wp--preset--spacing--20: 0.44rem;
	--wp--preset--spacing--30: 0.67rem;
	--wp--preset--spacing--40: 1rem;
	--wp--preset--spacing--50: 50px;
	--wp--preset--spacing--60: 60px;
	--wp--preset--spacing--70: 3.38rem;
	--wp--preset--spacing--80: 5.06rem;
	--wp--preset--spacing--10: 10px;
	--wp--preset--spacing--90: 90px;

	// Step 2
	--wp--preset--spacing--20: 0.44rem;
	--wp--preset--spacing--30: 0.67rem;
	--wp--preset--spacing--40: 1rem;
	--wp--preset--spacing--50: 1.5rem;
	--wp--preset--spacing--60: 2.25rem;
	--wp--preset--spacing--70: 3.38rem;
	--wp--preset--spacing--80: 5.06rem;
	--wp--preset--spacing--10: 10px;
	--wp--preset--spacing--90: 90px;
}
```

### Testing Instructions for Keyboard


## Screenshots or screencast
